addons/resource/models/resource.py

```diff
@@ -244,31 +216,27 @@ def get_leave_intervals(self, cr, uid, id, resource_id=None,
                                         ending before this date.
         :param datetime end_datetime: if provided, do not take into account leaves
                                         beginning after this date.
-
         :return list leaves: list of tuples (start_datetime, end_datetime) of
                              leave intervals
         """
-        resource_calendar = self.browse(cr, uid, id, context=context)
+        self.ensure_one()
         leaves = []
-        for leave in resource_calendar.leave_ids:
+        for leave in self.leave_ids:
```

Comment:
cannot we make use of filtered? like : 
for leave in self.leave_ids.filtered(lambda leave: leave.resource_id and resource_id == leave.resource_id.id):